Korelasyon Analizi Örneği

E-Ticaret şirketi X, belirli bir dönem boyunca her gün web sitesine gelen ziyaretçi sayısını ve bu ziyaretçilerin yapmış olduğu satın alma işlemlerinin sayısını kaydetmiştir. 

In [2]:
import pandas as pd
import numpy as np

In [3]:
data = {
    'day' : [1,2,3,4,5,6,7],
    'number of visitors' : [100,120,110,130,140,150,160],
    'number of sales' : [5,6,5,7,8,9,10]
}

In [5]:
df = pd.DataFrame(data)
df

day  number of visitors  number of sales
0    1                 100                5
1    2                 120                6
2    3                 110                5
3    4                 130                7
4    5                 140                8
5    6                 150                9
6    7                 160               10

Bu verilerle pearson korelasyon katsayısı hesaplayabiliriz. (Pearson korelasyon katsayısı, -1 ile 1 arasında değer alır. +1 tam pozitif korelasyon, -1 tam negatif korelasyonu temsil eder. 0: ilişki yok)

In [10]:
correlation = df['number of visitors'].corr(df['number of sales'])
print("Korelasyon katsayısı:", correlation)

Korelasyon katsayısı: 0.9882117688026183


Yorumlama: (Korelasyon katsayısı: 0.9882117688026183) korelasyon katsayısı 1'e yakın olduğu için çok güçlü pozitif korelasyon diyebiliriz. (bir değişken artarken diğeri de artıyor)